### ISSUE TITLE
Bug: `no-useless-default-assignment` inconsistent with `typescript-eslint`

### ISSUE THREAD

**Issue by atitcreate** [oxc-project/tsgolint#727]
cd to a temp dir, run:
```bash
npm install --save-dev eslint @eslint/js typescript typescript-eslint oxlint oxlint-tsgolint
npx tsc --init
```
create `eslint.config.mjs`:
```js
import eslint from "@eslint/js"
import { defineConfig } from "eslint/config"
import tseslint from "typescript-eslint"
export default defineConfig(eslint.configs.recommended, tseslint.configs.strictTypeChecked, {
  languageOptions: {
    parserOptions: {
      projectService: true,
    },
  },
})
```
create `test.ts`:
```ts
const config = { rules: {} }
// eslint-disable-next-line @typescript-eslint/no-unused-vars
const { rules = {} } = config
```
run `npx eslint test.ts`
result: `3:17  error  Default value is useless because the property is not optional  @typescript-eslint/no-useless-default-assignment`
run `npx oxlint --type-aware test.ts`
result: `Found 0 warnings and 0 errors.`

**Comment by camc314** [comment]
Can you share your oxlint config file?

https://oxc.rs/docs/guide/usage/linter/rules/typescript/no-useless-default-assignment.html#typescript-no-useless-default-assignment

The rule is in the nursery catergory which is not enabled by default.

To enable the rule supply the command line flag:
`-D no-useless-default-assignment`
Or configure your config file:
```json
{
    "rules": {
        "typescript/no-useless-default-assignment": "error"
    }
}
```


After making these changes and configuring oxlint does it work as expected? Or do you still see the issue

I've made a PR showing that we do report this case correctly

**Comment by atitcreate** [comment]
> The rule is in the nursery catergory which is not enabled by default.

Sorry I missed this(I encountered the issue with an oxlint config not enabling nursery catergory and reproduced the issue without config(use default config).
After enabling the rule explicitly oxlint does report the error.

**Comment by camc314** [comment]
Great - thanks.


> Sorry I missed this(I encountered the issue with an oxlint config not enabling nursery catergory and reproduced the issue without config(use default config).

is this a separate bug - do you mind making an issue in https://github.com/oxc-project/oxc/issues ?

**Comment by atitcreate** [comment]
> is this a separate bug - do you mind making an issue in https://github.com/oxc-project/oxc/issues ?

What is the "separate bug" and the "issue" to make?

**Comment by camc314** [comment]
You said _"I encountered the issue with an oxlint config not enabling nursery catergory"_

Is this a bug in oxc? If so could you make an issue with a reproduction so that it can be fixed?

if it's not a bug, then there's nothing to do here.

**Comment by atitcreate** [comment]
> Is this a bug in oxc? If so could you make an issue with a reproduction so that it can be fixed?
 
It's not a bug. I mean I didn't enable the nursery category in the oxlint config, and was under the impression that the rule didn't take effect.

**Comment by Smrtnyk** [comment]
in order to enable one nursery rule explicitely you had to enable the whole nursery category?

**Comment by atitcreate** [comment]
> in order to enable one nursery rule explicitely you had to enable the whole nursery category?

You don't need to. [You can only enable a specific rule as the reply above shows](#issuecomment-3941205324)
